pygdf/datetime.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from libgdf_cffi import ffi, libgdf
+import weakref
+from . import _gdf, columnops, utils, cudautils
+from .buffer import Buffer
+from .cudautils import compact_mask_bytes
+
+
+class DatetimeColumn(columnops.TypedColumnBase):
+    def __init__(self, data, mask=None, null_count=None, dtype=None):
+        # currently libgdf datetime kernels fail if mask is null
+        if mask is None:
+            mask = np.empty(data.mem.size, dtype=np.bool)
+            mask[:] = True
+            mask = compact_mask_bytes(mask)
+            mask = Buffer(mask)
+        super(DatetimeColumn, self).__init__(data=data,
+                                             mask=mask,
+                                             null_count=null_count,
+                                             dtype=dtype
+        )
+        # the column constructor removes mask if it's all true
+        self._mask = mask
+
+
+
+funcs = {
```

Comment:
This seems like it should be a class attribute of DatetimeProperties, rather than in the global namespace.